From 67c0427180e30d0a372d3bde563c31ec099c8e18 Mon Sep 17 00:00:00 2001
From: Conor McCarthy <cmccarthy@codeweavers.com>
Date: Tue, 16 Jul 2019 13:41:08 +1000
Subject: [PATCH] dbghelp: Return NULL instead of IMAGE_NO_MAP when PE file
 mapping fails.

A NULL test is always applied to the return value from pe_map_full(), but
IMAGE_NO_MAP is not NULL, it is ((void*)-1).

Signed-off-by: Conor McCarthy <cmccarthy@codeweavers.com>
---
 dlls/dbghelp/pe_module.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/dbghelp/pe_module.c b/dlls/dbghelp/pe_module.c
index 58530e7343..4b7c7baaae 100644
--- a/dlls/dbghelp/pe_module.c
+++ b/dlls/dbghelp/pe_module.c
@@ -53,7 +53,7 @@ static void* pe_map_full(struct image_file_map* fmap, IMAGE_NT_HEADERS** nth)
         fmap->u.pe.full_count++;
         return fmap->u.pe.full_map;
     }
-    return IMAGE_NO_MAP;
+    return NULL;
 }
 
 static void pe_unmap_full(struct image_file_map* fmap)
-- 
2.17.1

